static/src/javascripts/projects/common/modules/onward/breaking-news.js

```diff
@@ -32,137 +32,197 @@ define([
-                        });
-
-                        fastdom.write(function () {
-                            $breakingNews.append(el);
-                        });
-
-                        if (hiddenIds[article.id] === false) {
-                            alertDelay = 0;
-                        }
-                    });
-
-                    setTimeout(function () {
-                        var message = 'breaking news alert shown' + (alertDelay ? '' : ' 2 or more times'), $breakingNewsSpectre;
-
-                        // copy of breaking news banner (with blank content) used inline in the body
-                        // to create space for a pinned alert to be scrolled into
-                        $body = $body || bonzo(document.body);
-                        $breakingNewsSpectre = bonzo(bonzo.create($breakingNews[0])).addClass('breaking-news--spectre').removeClass('breaking-news--hidden');
-                        fastdom.write(function () {
-                            $body.append($breakingNewsSpectre);
-                        });
-
-                        if (!alertDelay) {
-                            fastdom.write(function () {
-                                $breakingNews.removeClass('breaking-news--fade-in');
-                            });
-                        }
-
-                        fastdom.write(function () {
-                            $breakingNews.removeClass('breaking-news--hidden');
-                        });
-
-                        omniture.trackLink(this, message);
-                    }, alertDelay);
-                }
-            }
-        );
-    };
+        });
+    }
+
+    // handle the breaking news JSON
+    function parseResponse(response) {
+        return (response.collections || [])
+            .filter(function (collection) {
+                return isArray(collection.content) && collection.content.length;
+            })
+            .map(function (collection) {
+                // collection.href is string or null
+                collection.href = (collection.href || '').toLowerCase();
+                return collection;
+            });
+    }
+
+    // pull out the alerts from the edition/section buckets that apply to us
+    // global > current edition > current section
+    function getRelevantAlerts(alerts) {
+        var edition = (page.edition || '').toLowerCase(),
+            section = supportedSections[page.section];
+
+        return flatten([
+            alerts
+                .filter(function (alert) {return alert.href === 'global';})
+                .map(function (alert) {return alert.content;}),
+            alerts
+                .filter(function (alert) {return alert.href === edition;})
+                .map(function (alert) {return alert.content;}),
+            alerts
+                .filter(function (alert) {return section && alert.href === section;})
+                .map(function (alert) {return alert.content;})
+        ]);
```

Comment:
seems like you could get rid of `flatten` altogether with one `filter` and a condition that combines those three checks